packages/batch-submitter/src/batch-submitter/batch-submitter.ts

```diff
@@ -17,12 +18,21 @@ export interface ResubmissionConfig {
   gasRetryIncrement: number
 }
 
+interface BatchSubmitterMetrics {
+  batchSubmitterETHBalance: Gauge<string>
+  batchSizeInBytes: Histogram<string>
+  numTxPerBatch: Histogram<string>
+  submissionTimestamp: Histogram<string>
+  submissionGasUsed: Histogram<string>
+}
+
 export abstract class BatchSubmitter {
   protected rollupInfo: RollupInfo
   protected chainContract: Contract
   protected l2ChainId: number
   protected syncing: boolean
   protected lastBatchSubmissionTimestamp: number = 0
```

Comment:
Could setting this to 0 potentially lead to edge cases as opposed to explicitly leaving it `undefined` until we have fetched this information?

Reply:
^i think undefined might leave more edge cases since we're trying to do a subtraction!

Reply:
Right I guess it's just a matter of how we prefer to treat "un-fetched data". If we just treat it as 0 then we check for 0 in the code, otherwise we check for undefined. TypeScript has some typing around undefined values being potentially operated on, which is why I recommended undefined, but I'm not super opionated.